File: app/src/main/java/com/parametris/iteng/asdf/comm/IRCListener.java
package com.parametris.iteng.asdf.comm;

import org.pircbotx.Configuration;
import org.pircbotx.hooks.ListenerAdapter;
import org.pircbotx.hooks.types.GenericMessageEvent;

public class IRCListener extends ListenerAdapter {
    @Override
    public void onGenericMessage(GenericMessageEvent messageEvent) {

    }

    public static void main(String[] args) throws Exception {
        Configuration configuration = new Configuration.Builder()
                .setName("TODO")
                .addListener(new IRCListener())
                .buildConfiguration();


    }
}

File: app/src/main/java/com/parametris/iteng/asdf/comm/IRCListener.kt
package com.parametris.iteng.asdf.comm

import org.pircbotx.Configuration.*
import org.pircbotx.hooks.ListenerAdapter
import org.pircbotx.hooks.types.GenericMessageEvent

class IRCListener : ListenerAdapter() {
    override fun onGenericMessage(messageEvent: GenericMessageEvent?) {

    }

    companion object {
        @Throws(Exception::class)
        @JvmStatic fun main(args: Array<String>) {
            val configuration = Builder().setName("TODO").addListener(IRCListener()).buildConfiguration()
        }
    }
}
